bazel/asn1c.bzl

```diff
@@ -173,13 +164,13 @@ def cc_asn1_library(
         flags = flags,
         prefix = prefix,
         substitutions = substitutions,
-        # We want to remove converter-example.c as it does not compile
-        files_to_remove = ["converter-example.c"],
     )
 
     cc_library(
         name = name,
         srcs = [gen_name],
         # This is needed so that the CCInfo (header/include info) can be used
         deps = [gen_name],
+        # TODO: add a comment here about why this is importatnt
+        linkstatic = True,
```

Comment:
I realize this is a draft and you are experimenting! But does this just result in only a static as1n_16.a output? Which then forces google test targets to static link? Is that ~normal? Maybe it is. I thought perhaps the dynamic linking of google test against target was for speed of compilation / linking / execution tests once you have built the target.

Reply:
I did 7 minutes of google-fu and I cannot find advisement on Bazel / cc_test on the merits of dynamic vs static linking for the target libraries...

Reply:
@electronjoe  Yes, so this would make it so that .a files are produced for asn library (https://docs.bazel.build/versions/main/be/c-cpp.html#cc_library.linkstatic)
```
The linkstatic attribute has a different meaning if used on a [cc_library()](https://docs.bazel.build/versions/main/be/c-cpp.html#cc_library) rule. For a C++ library, linkstatic=True indicates that only static linking is allowed, so no .so will be produced. linkstatic=False does not prevent static libraries from being created. The attribute is meant to control the creation of dynamic libraries.
```

I am having difficulty understanding why dynamic linking is broken for the asn1 library, so I thought this might be an OK workaround as it only forces static linking for the ASN libraries.

Reply:
Setting `linkstatic=True` in cc_binary (True by default) or cc_test (False by default) would force all dependencies to be statically linked. Setting `linkstatic=True` in cc_library forces that particular library to be always statically linked. (Which is what you are saying above, I think)

Reply:
I am still trying to understand why the dynamic linking is broken for the custom rule though... But I haven't made much progress there yet.

Reply:
@themarwhal and I discussed and (as she already realized, but I just came to realize) forcing linkstatic=True for just asn1c is not nearly such a large penalty as my prototype solution to this issue (which was to set linkstatic=True for all MME tests - which impacts far more libraries / slows things down a lot).

TL;DR; your approach to force static linking for asn1c sounds OK to me at this point.